versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/MergeSquashImpl.java

```diff
@@ -52,7 +52,7 @@ final class MergeSquashImpl extends BaseMergeTransplantSquash implements Merge {
   public MergeResult<Commit> merge(Optional<?> retryState, MergeOp mergeOp)
       throws ReferenceNotFoundException, RetryException, ReferenceConflictException {
     ObjId fromId = hashToObjId(mergeOp.fromHash());
-    ObjId commonAncestorId = identifyCommonAncestor(fromId);
+    ObjId commonAncestorId = identifyMergeBase(fromId).id();
 
     SourceCommitsAndParent sourceCommits = loadSourceCommitsForMerge(fromId, commonAncestorId);
```

Comment:
Not sure this will work correctly when the merge base is not a direct ancestor... `loadSourceCommitsForMerge` appears to assume that both hashes are part of the predecessor-based commit log.

Reply:
Ouch - yes.
As discussed offline, the commit-meta information will no longer be constructed from the "merged commits", because that would become a real PITA wrt to more complex merge scenarios (see the `TestMergeBase` test cases).

All new commits are there to serve the above purpose:
* the actual production code changes are relatively simple
* there are a bunch of necessary simplification and cleanup changes
* the necessary test code changes make the tests quite complex wrt to the test scenarios and data model being tested - #6890 will eventually clean this m... up